Watchman.Integrations/MongoDB/Entity.cs

```diff
@@ -8,6 +8,7 @@ public abstract class Entity
         public DateTime CreatedAt { get; protected set; } = DateTime.UtcNow;
         public DateTime UpdatedAt { get; protected set; } = DateTime.UtcNow;
         public int Version { get; protected set; }
+        public bool IsDeleted { get; set; }
```

Comment:
tego nie powinno się robić bezpośrednio przez setter
lepiej zrobić metodę
protected void Delete()

w sumie myślę jeszcze czy nie warto również inkrementować this.Version
@Marcin99b co myślisz?

Reply:
potwierdzam
dodatkowo metoda Delete() powinna odpalać metode Update() żeby nadpisać UpdatedAt itd

Reply:
no i ogólnie w obiektach domenowych nic nie powinno mieć publicznego settera
bo chcemy żeby każda zmiana aktualizowała UpdatedAt i Version 

w przyszłości fajnie byłoby jeszcze zapisywać co kiedy zmieniło sie z czego na co
ale to tylko takie "fajnie by było" bo nie jest to dla nas kluczowe a sporo miejsca by zajmowało w bazie